core/main/src/Utils/ColorUtils.ts

```diff
@@ -479,16 +479,33 @@ export class ColorUtils {
         };
     }
 
-    static replaceColorSvg(image: IImage, color: IHsl, opacity: number): string {
-        if (!image.svgData) {
-            return "";
+    static replaceColorSvg(imageShape: IImage, color: IHsl, opacity: number): string {
+        if (!imageShape) {
+            throw new Error("Error tsParticles - No image provided.");
+        }
+
+        const { source, svgData } = imageShape;
+        if (!source || !svgData) {
+            throw new Error("Error tsParticles - No image.src");
+        }
+
+        const imageType = source.substr(source.length - 3);
+        if (imageType !== "svg") {
+            throw new Error("Error tsParticles - replaceColor property provided to non-.svg image.src");
         }
 
         /* set color to svg element */
-        const svgXml = image.svgData;
-        const rgbHex = /#([0-9A-F]{3,6})/gi;
+        if (svgData.includes("fill")) {
+            const currentColor = /(#(?:[0-9a-f]{2}){2,4}|(#[0-9a-f]{3})|(rgb|hsl)a?\((-?\d+%?[,\s]+){2,3}\s*[\d.]+%?\)|currentcolor)/i;
```

Comment:
Originally I indented to use a much larger [regex](https://gist.github.com/olmokramer/82ccce673f86db7cda5e) that also encompasses color names like "black, orange, pink" etc. But I didn't know where to put it (which file to export it from) so I used this one here which is much shorter, and as for string names, I left only "currentcolor" since I noticed it's the default value on [Boostrap Icons](https://icons.getbootstrap.com/)

Reply:
`currentcolor` is not a valid value for `getStyleFromHsl`, I think it can be ignored